sdk/cosmosdb/cosmos/test/public/integration/retry.spec.ts

```diff
@@ -1,144 +1,187 @@
 // Copyright (c) Microsoft Corporation.
 // Licensed under the MIT License.
 
-import { describe, it } from "vitest";
+import { describe, it, beforeAll } from "vitest";
+import { CosmosClient, type CosmosClientOptions } from "../../../src/index.js";
+import { endpoint } from "../common/_testConfig.js";
+import { masterKey } from "../common/_fakeTestSecrets.js";
+import { addEntropy } from "../common/TestHelpers.js";
+import assert from "assert";
 
-// import assert from "assert";
-// import { ConnectionPolicy, Constants, CosmosClient, RetryOptions } from "../../src";
-// import * as request from "../../src/request";
+const isBrowser = new Function("try {return this===window;}catch(e){ return false;}");
 
-describe("retry policy tests", () => {
-  // this.timeout(300000);
-  // const collectionDefinition = {
-  //   id: "sample collection"
-  // };
+if (!isBrowser()) {
+  describe("HTTP Proxy Retry Policy Integration Tests", () => {
+    let testDatabaseId: string;
 
-  // const documentDefinition = {
-  //   id: "doc",
-  //   name: "sample document",
-  //   key: "value"
-  // };
+    beforeAll(async () => {
+      testDatabaseId = addEntropy("RetryProxyTest");
+    });
 
-  // mocked database account to return the WritableLocations and ReadableLocations
-  // set with the default endpoint
-  // const mockGetDatabaseAccount = function (options, callback) {
-  //     const databaseAccount = new AzureDocuments.DatabaseAccount();
-  //     callback(undefined, databaseAccount);
-  // };
+    const createTestClient = (options: Partial<CosmosClientOptions> = {}): CosmosClient => {
+      return new CosmosClient({
+        endpoint,
+        key: masterKey,
+        ...options,
+        connectionPolicy: {
+          enableBackgroundEndpointRefreshing: false,
+          ...options.connectionPolicy,
+        },
+      });
+    };
 
-  // const retryAfterInMilliseconds = 1000;
-  // // mocked request object stub that calls the callback with 429 throttling error
-  // const mockCreateRequestObjectStub = function (connectionPolicy, requestOptions, callback) {
-  //     callback({ code: 429, body: "Request rate is too large",
-  //                retryAfterInMilliseconds: retryAfterInMilliseconds });
-  // };
+    it("throttle retry policy test default retryAfter", async () => {
+      const collectionDefinition = { id: "sample-collection" };
+      const documentDefinition = { id: "doc", name: "sample document", key: "value" };
+      const retryOptions = { maxRetryAttemptCount: 5 };
+      const client = createTestClient({ connectionPolicy: { retryOptions } });
+      const db = await client.databases.create({ id: testDatabaseId + "_db1" });
+      const { container } = await client
+        .database(db.database.id)
+        .containers.create(collectionDefinition);
+      try {
+        await client
+          .database(db.database.id)
+          .container(container.id)
+          .items.create(documentDefinition);
+      } catch (err: any) {
```

Comment:
Where are we mocking 429s?

Reply:
Is this catch block (and the subsequent tests catch block) even getting triggered?